### PR TITLE
adding game tick rules

### PR DESCRIPTION
**Pull request by dlsteuer**
started putting in game tick logic

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -105,6 +105,15 @@ func (s *Server) Create(ctx context.Context, req *pb.CreateRequest) (*pb.CreateR
 	}, nil
 }
 
+// Update updates the same state in the data store
+func (s *Server) Update(ctx context.Context, req *pb.UpdateRequest) (*pb.UpdateResponse, error) {
```
[the remark this answers is not in the document]
Comment (dlsteuer) [reply controller/controller.go]: yeah it's a thought, let me think about it

[the remark this answers is not in the document]
Comment (dlsteuer) [reply controller/controller.go]: one big issue is the protobuf generated structs aren't immutable, so we'd have to use our own structs, but maybe that's not so bad.

[the remark this answers is not in the document]
Comment (colin-swu) [reply controller/controller.go]: I just mean not providing the `Update` call, I think it's ok to have mutable representations of immutable data. Pointers are more efficient sometimes.

[the remark this answers is not in the document]
Comment (dlsteuer) [reply controller/controller.go]: but then how do you propose saving the game back in the store? The "update" just puts the entire game back into the store

[the remark this answers is not in the document]
Comment (dlsteuer) [reply controller/controller.go]: So thinking about this, this morning, if you wanted to add an `AddGameTick` controller method that works for me too.

**Pull request merged**